addons/test_mail/tests/test_mail_activity.py

```diff
@@ -148,6 +149,35 @@ def test_activity_dont_notify_no_user_change(self):
             activity.with_user(self.user_admin).write({'user_id': self.user_employee.id})
         self.assertEqual(activity.user_id, self.user_employee)
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_activity_summary_on_change(self):
+        ActivityType = self.env['mail.activity.type']
+        email_activity_type = ActivityType.create({
+            'name': 'email',
+            'summary': 'Email Summary'
+            })
+        call_activity_type = ActivityType.create({'name': 'call'})
+        review_activity_type = ActivityType.create({'name': 'review'})
```

Comment:
You can limit the test's footprint using 2 activity types (one with summary, one without).